apps/04TowersOfHanoi.js

```diff
@@ -10,37 +10,85 @@ var stacks = {
     c: []
 };
 
+var counter = 0
+function add() {
+  counter++;
+}
+
+
 function printStacks() {
     console.log("a: " + stacks.a);
     console.log("b: " + stacks.b);
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
-
+function movePiece(startStack, endStack) {
+  var piece = stacks[startStack].pop();
+  stacks[endStack].push(piece);
 }
 
-function isLegal() {
-    // Your code here
-
+function isLegal(startStack, endStack) {
+    var startStackArray = stacks[startStack];
+    var endStackArray = stacks[endStack];
+    if (startStackArray.length === 0) {
+      return false;
+    }
+    else if (endStackArray.length === 0) {
+      return true;
+    }
+    else {
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
+    }
 }
 
 function checkForWin() {
-    // Your code here
-
+    if (stacks.b.length === 4 || stacks.c.length === 4 ) {
```

Comment:
Remember `if (cond) { return true; } else { return false; }` can always be improved/shortened